### DISCUSSION TITLE
How to activate virtual environment while running inside GitHub actions ?

### DISCUSSION THREAD

**Discussion by aahnik** [python-poetry/poetry#3966] [Q&A]
![image](https://user-images.githubusercontent.com/66209958/115117053-5a246480-9fba-11eb-9a73-17254e4bbce5.png)

**Comment by BD103** [comment]
Spawning a virtual environment shouldn't be necessary for Github Actions, as it is already a virtual environment. You should be able to use Poetry without running `poetry shell`.

**Reply by aahnik** [reply]
that's ok. but @abn advised me to run inside virtual environments always. I am currently doing `poetry run mycommand`

link to @abn advice https://github.com/python-poetry/poetry/discussions/3946#discussioncomment-616632
